### PR TITLE
feat(query): add and_filters function

### LINKED ISSUE #5382: Feature: Efficient way to combine multiple filters
Do we have a efficient way to combine multiple filter columns(i.e. `BooleanArray`), like `combine_validities`？

cc @sundy-li

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

The default `and`, `or` behavior is following as https://en.wikipedia.org/wiki/Three-valued_logic#Kleene_and_Priest_logics .
The behavior is compatible with pg and mysql.

In order to support fast filters combination, we introduce another function named `and_filters` to combine the filters.

Fixes #5382

### CODE REVIEW

**Comment by sundy-li** [comment]
@mergify update

File: src/query/functions-v2/tests/it/scalars/testdata/boolean.txt [hunk 1]
```diff
@@ -7,15 +7,51 @@ output domain  : {FALSE}
...
+
+ast            : false AND null
+raw expr       : and(false, NULL)
+checked expr   : and<Boolean NULL, Boolean NULL>(CAST(false AS Boolean NULL), CAST(NULL AS Boolean NULL))
 optimized expr : NULL
-output type    : NULL
-output domain  : {NULL}
+output type    : Boolean NULL
+output domain  : Unknown
 output         : NULL
```
Comment (TCeason) [inline src/query/functions-v2/tests/it/scalars/testdata/boolean.txt:43]: Now the false and null will return false. Why we modify this?

Comment (sundy-li) [reply src/query/functions-v2/tests/it/scalars/testdata/boolean.txt:43]: See the description in this pr.

File: src/query/expression/src/types/nullable.rs [hunk 2]
```diff
@@ -69,6 +69,14 @@ impl<T: ValueType> ValueType for NullableType<T> {
                 has_null: *has_null,
                 value: Some(Box::new(T::try_downcast_domain(value)?)),
             }),
+
```
Comment (andylokandy) [inline src/query/expression/src/types/nullable.rs:72]:
```suggestion
```

**Review by andylokandy — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]
+1

**Review by TCeason — APPROVED** [review]
LGTM

**Pull request merged**